### PR TITLE
Remove dependency for integration framework tests from e2e fw

### PR DESCRIPTION
**Pull request by gavinfish**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Remove integration tests framework dependency on e2e test framework.

**Which issue(s) this PR fixes**:

Xref #89185

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
none
```

### CODE REVIEW

File: test/integration/framework/util.go [hunk 1]
```diff
@@ -43,3 +64,227 @@ func CreateTestingNamespace(baseName string, apiserver *httptest.Server, t *test
...
+	}
+	if !silent {
+		klog.Infof("Couldn't find condition %v on node %v", conditionType, node.Name)
+	}
+	return false
+}
+
+// isNodeUntainted tests whether a fake pod can be scheduled on "node", given its current taints.
+// TODO: need to discuss wether to return bool and error type
+func isNodeUntainted(node *v1.Node) bool {
```
Comment (oomichi) [inline test/integration/framework/util.go]: Why do we need to implement `isNodeUntaintedWithNonblocking()`?
The function is only called from this function, and the above seems unnecessary wrapper.
We can change like
```
- func isNodeUntaintedWithNonblocking(node *v1.Node, nonblockingTaints string) bool {
+ func isNodeUntainted(node *v1.Node) bool {
+     nonblockingTaints  := ""
```

Comment (gavinfish) [reply test/integration/framework/util.go]: Make sense. Honestly, the changes in this PR make me uncomfortable. You know, lots of duplicated code across the codebase, especially with lots of TODO items. Maybe we should rethink about this one https://github.com/kubernetes/kubernetes/blob/42972f2a37c56b5f850086a7330b34ae6592f8ac/test/integration/framework/util.go#L17.

Comment (tanjunchen) [reply test/integration/framework/util.go]: @gavinfish 
Many TODOs cannot be traced to their causes or historical background due to code changes, movements, and refactorings. Welcome to solve some of them, haha
see  https://github.com/kubernetes/kubernetes/issues/86763  https://github.com/kubernetes/kubernetes/issues/86756 
welcome

**Review by oomichi — COMMENTED** [review]
Thanks for updating,

/lgtm

File: test/integration/framework/util.go (hunk 1 above)
Comment (andrewsykim) [reply test/integration/framework/util.go]: > Make sense. Honestly, the changes in this PR make me uncomfortable. You know, lots of duplicated code across the codebase, especially with lots of TODO items. Maybe we should rethink about this one 

I agree with this. Little duplication here and there is okay but this is a big one. Wondering if there's a cleaner way to do this.

Comment (gavinfish) [reply test/integration/framework/util.go]: @andrewsykim I have created #89292 to track this issue.

**Pull request merged**